plugins/package-managers/go/src/main/kotlin/GoMod.kt

```diff
@@ -104,6 +104,8 @@ class GoMod(
             val cleanGoPath = createOrtTempDir()
             put("GOPATH", cleanGoPath.absolutePath)
             put("GOPROXY", "direct")
+        } else if (!config.disableGoProxy && System.getenv("GOPROXY") != null) {
+            put("GOPROXY", System.getenv("GOPROXY"))
```

Comment:
Some comments here:
- The `!config.disableGoProxy &&` part is not necessary.
- While I get what you try to do here, the original intent of `disableGoProxy` was really do *disable* the Go proxy.

Let me think about this a bit and make a proposal for improvement.

Reply:
I've checked the code again, and I believe your changes should not be necessary. The code at

https://github.com/oss-review-toolkit/ort/blob/60316b797569f05fc7835f6e2f3b2bb72340f71b/plugins/package-managers/go/src/main/kotlin/GoMod.kt#L111

only specified *additional* environment variables use, on top of those available to the environment running ORT. So any set `GOPROXY` variable would get inherited.

Can you confirm that this is the case, @losmateusz1?

Anyway, I agree that the current option name is misleading, as it does not completely disable any set `GOPROXY`. I'm clarifying that in https://github.com/oss-review-toolkit/ort/pull/11189.